package/src/plugins/has-vite-plugin.ts

```diff
@@ -0,0 +1,58 @@
+import type { AstroConfig } from "astro";
+import type { PluginOption } from "vite";
+import { definePlugin } from "../core/define-plugin.js";
+import { hasVitePlugin } from "../utilities/has-vite-plugin.js";
+
+function getPlugins(
+	store: Set<
+		Extract<
+			NonNullable<NonNullable<AstroConfig["vite"]["plugins"]>[number]>,
+			{ name: string }
+		>
+	>,
+	plugins: AstroConfig["vite"]["plugins"],
+) {
+	if (plugins) {
+		for (const plugin of plugins) {
+			if (!plugin) continue;
+
+			if (Array.isArray(plugin)) {
+				getPlugins(store, plugin);
+				continue;
+			}
+
+			if (plugin instanceof Promise) {
+				continue;
+			}
+
+			store.add(plugin);
+		}
+	}
+	return store;
+}
+
+export const hasVitePluginPlugin = definePlugin({
+	name: "hasVitePlugin",
+	hook: "astro:config:setup",
+	implementation: (astroConfig) => {
+		const currentPlugins = getPlugins(
+			new Set(),
+			astroConfig.config.vite?.plugins,
+		);
+
+		const { updateConfig } = astroConfig;
+
+		astroConfig.updateConfig = (config) => {
```

Comment:
Would this break for multiple integrations both using AIK and `hasVitePlugin` ? Would the second `astroConfig.updateConfig` overwrite this one? I have absolutely no idea mind

Reply:
Each integration receives its own `updateConfig`

Reply:
Ya this is scoped to each call of `defineIntegration` so nothing gets shared between integrations